packages/helper-wasm-section/src/resize.js

```diff
@@ -33,7 +33,29 @@ export function resizeSectionByteSize(
    * update AST
    */
   sectionMetadata.size.value = newSectionSize;
-  sectionMetadata.size.loc.end.column = start + newBytes.length;
+
+  const oldu32EncodedLen = end - start;
+  const newu32EncodedLen = newBytes.length;
+
+  // For padded LEB128 we can end up resizing the section again because we don't
+  // use the same encoding (thus number of bytes)
+  if (newu32EncodedLen !== oldu32EncodedLen) {
+    const deltaInSizeEncoding = newu32EncodedLen - oldu32EncodedLen;
+
+    sectionMetadata.size.loc.end.column = start + newu32EncodedLen;
```

Comment:
wont the end be, `end + newu32EncodedLen`

Reply:
This is actually the end of the `size` of the section number.